ethereum/eth/src/main/java/org/hyperledger/besu/ethereum/eth/transactions/TransactionPoolMetrics.java

```diff
@@ -97,6 +102,14 @@ public TransactionPoolMetrics(final MetricsSystem metricsSystem) {
             "The number of transactions currently present in the layer",
             "layer");
 
+    transactionCountByType =
+        metricsSystem.createLabelledGauge(
+            BesuMetricCategory.TRANSACTION_POOL,
+            "number_of_transactions_by_type",
+            "The number of transactions, of a specified type, currently present in the layer",
+            "layer",
+            "type");
```

Comment:
In addition to the type name, should we expose the EIP-2718 type number?  We shouldn't use ordinal and should encode it in TransactionType but that has some weirdness for frontier types.  May be best for a future PR.

Reply:
yes I think it makes sense to explicitly add the EIP-2718 type number to the tx type, and expose it in the metric label.
About the use of `type.ordinal()`, it is only used internally to index the array of int for the counters, and not exposed outside.